instaclustr/resource_cluster.go

```diff
@@ -325,29 +365,165 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
+		if bundles[i].Bundle == "KAFKA_REST_PROXY" {
+			hasRestProxy = true
+		}
+		if bundles[i].Bundle == "KAFKA_SCHEMA_REGISTRY" {
+			hasSchemaRegistry = true
+		}
+	}
+
+	if len(waitForClusterState) > 0 {
+		return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+			cluster, err := client.ReadCluster(id)
+
+			if err != nil {
+				return resource.NonRetryableError(fmt.Errorf("[Error] Error retrieving cluster info: %s", err))
+			}
+
+			if cluster.ClusterStatus == waitForClusterState {
+
+				if isKafkaCluster {
+
+					if hasRestProxy {
+						// preparing the kafka rest proxy bundle user update request
+						updateRestProxyData := UpdateBundleUserRequest{
+							Username: "ickafkarest",
+							Password: d.Get("kafka_rest_proxy_user_password").(string),
+						}
+						var jsonStrUpdateRestProxy []byte
+						jsonStrUpdateRestProxy, err := json.Marshal(updateRestProxyData)
+
+						if err != nil {
+							return resource.NonRetryableError(fmt.Errorf("[Error] Error creating the kafka rest proxy bundle user update request : %s", err))
+						}
+
+						//updating the kafka rest proxy bundle user
+						if len(kafkaRestProxyUserPassword) > 0 {
+							err = client.UpdateBundleUser(d.Get("cluster_id").(string), "kafka_rest_proxy", jsonStrUpdateRestProxy)
+						}
+						if err != nil {
+							return resource.NonRetryableError(fmt.Errorf("[Error] Error updating the kafka rest proxy bundle user password : %s", err))
+						}
+					}
+
+					if hasSchemaRegistry {
+						// preparing the kafka schema registry bundle user update request
+						updateSchemaRegistryData := UpdateBundleUserRequest{
+							Username: "ickafkaschema",
+							Password: d.Get("kafka_schema_registry_user_password").(string),
+						}
+						var jsonStrUpdateSchemaRegistry []byte
+						jsonStrUpdateSchemaRegistry, err = json.Marshal(updateSchemaRegistryData)
+
+						if err != nil {
+							return resource.NonRetryableError(fmt.Errorf("[Error] Error creating the kafka schema registry bundle user update request : %s", err))
+						}
+
+						//updating the kafka schema registry bundle user
+						if len(kafkaSchemaRegistryUserPassword) > 0 {
+							err = client.UpdateBundleUser(d.Get("cluster_id").(string), "kafka_schema_registry", jsonStrUpdateSchemaRegistry)
+						}
+						if err != nil {
+							return resource.NonRetryableError(fmt.Errorf("[Error] Error updating the kafka schema registry bundle user password : %s", err))
+						}
+					}
+
+					return resource.NonRetryableError(resourceClusterRead(d, meta))
+				} else if !isKafkaCluster && (len(kafkaSchemaRegistryUserPassword) > 0 || len(kafkaRestProxyUserPassword) > 0) {
+					return resource.NonRetryableError(fmt.Errorf("[Error] Error updating the bundle user passwords, because it should be a KAFKA cluster in order to update the schema-registry or rest-proxy users"))
+				}
+			}
+
+			return resource.RetryableError(fmt.Errorf("[Error] Cluster is in state %s", cluster.ClusterStatus))
```

Comment:
Suggest to change the error string here to -

`fmt.Errorf("[DEBUG] Cluster is in state %s, waiting for it to reach state %s", cluster.ClusterStatus, waitForClusterState)`